### DISCUSSION TITLE
Model Benchmarks in Cline

### DISCUSSION THREAD

**Discussion by Bigsy** [cline/cline#1013] [Feature Requests]
A bit like what Aider does. Everyone knows Cline works well with Sonnet but its expensive and beyond that it pretty unknown what other models work well and how close they are to sonnet. I think it would possibly drive more adoption.

1. People know what cheaper alternatives will work well under what conditions (diff/whole file edits etc).
2. Social media content, people love reading how good the latest models are at coding - Aider seem to get good publicity from it.
3. Stop the (well mostly) endless question about the same topic.

**Comment by elibroftw** [comment]
Honestly. With the release of DeepSeek v3, DeepSeek R1, QwQ 32B Preview, Google Flash 2.0 Thinking, I thought [LiveCodeBench](https://livecodebench.github.io/leaderboard.html) would be relevant but nope. Claude 3.5 outperforms them all when it comes to using Cline pair programming (that's what I'm calling it). Here's my take. I'm trying to avoid Reddit so that's why I'm commenting here.

- DeepSeek R1: API is very slow. The response takes forever and I end up cancelling after 30 seconds. I'd rather code it myself if it's going to take that long. I don't use AI for thinking, I use it for donkey work.
- Deepseek V3: noticeably slower than Claude, but probably a tenth of the cost
- QwQ 32B Preview: a lot of yapping, not a lot of coding. This model is a parody of people in real life who talk a lot but don't get any work done.
- Gemini 2.0 Flash Thinking Experimental 01-21: API rate limit exceeded. I'm happy to pay, so the fact that this is free is actually a bad sign. Is it Google's fault?

I'm probably going to start making SWE AI LLM Recap videos on a monthly basis because I have the time to explore models when coding. Will probably need to test out o1 as well.
